ocs_ci/utility/cco.py

```diff
@@ -0,0 +1,217 @@
+"""
+Cloud Credential Operator utility functions
+"""
+import logging
+import os
+import shutil
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import (
+    delete_file,
+    download_file,
+    exec_cmd,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def configure_cloud_credential_operator():
+    """
+    Extract and Prepare the CCO utility (ccoctl) binary. This utility
+    allows us to create and manage cloud credentials from outside of
+    the cluster while in manual mode.
+
+    """
+    bin_dir = config.RUN["bin_dir"]
+    ccoctl_path = os.path.join(bin_dir, "ccoctl")
+    if not os.path.isfile(ccoctl_path):
+        # retrieve ccoctl binary from https://mirror.openshift.com
+        version = config.DEPLOYMENT.get("ccoctl_version")
+        source = f"https://mirror.openshift.com/pub/openshift-v4/clients/ocp/{version}/ccoctl-linux.tar.gz"
+        bin_dir = config.RUN["bin_dir"]
+        tarball = os.path.join(bin_dir, "ccoctl-linux.tar.gz")
+        logger.info("Downloading ccoctl tarball from %s", source)
+        download_file(source, tarball)
+        cmd = f"tar -xzC {bin_dir} -f {tarball} ccoctl"
+        logger.info("Extracting ccoctl binary from %s", tarball)
+        exec_cmd(cmd)
+        delete_file(tarball)
+
+
+def get_release_image(openshift_installer):
+    """
+    Retrieve release image using the openshift installer.
+
+    Args:
+        openshift_installer (str): Path to the openshift installer
+
+    Returns:
+        str: Release image from the openshift installer.
+
+    """
+    logger.info("Retrieving release image")
+    cmd = f"{openshift_installer} version"
+    proc = exec_cmd(cmd)
+    for line in proc.stdout.decode().split("\n"):
+        if "release image" in line:
+            return line.split(" ")[2].strip()
```

Comment:
There is [a similar function in ocs_ci/deployment/disconnected.py](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/disconnected.py#L552-L571), so it might be worth to check it and prospectively unify it (and move to some more common place) - if it would make sense.

Reply:
Good catch! I moved both iteration of this to the deployment utility module and refactored the one used in `disconnected` a bit. I wasn't 100% sure if the `custom_ocp_image` logic was something we wanted for AWS IPI with STS or IBM Cloud IPI so I didn't fully combine the two functions. If we want that logic being used for those scenarios however I can update the PR and fully combine them into the same function call.